mec/pkg/model/interfaces.go

```diff
@@ -14,9 +14,13 @@
 
 package model
 
+import (
+	corev1 "k8s.io/api/core/v1"
+)
+
 type ImagePullStrategy interface {
 	// PullImage will go and Pull and image from a remote registry
-	PullImage(image *ImageRef) (Image, error)
+	PullImage(image *ImageRef, namespace string, pullPolicy corev1.PullPolicy, pullSecrets []corev1.LocalObjectReference) (Image, error)
```

Comment:
why move the handling of the `pullPolicy` into the `ImagePullStrategy`? shouldn't the worker make the decision based on status of the resource + return value of `strategy.GetImage()`?

Reply:
GetImage() only works if the image is in the format registry/repo@sha256... which is not the case in most cases. In fact, when the user uses that format, pullPolicy doesn't matter.

Reply:
you're right, I forgot that the pullPolicy is relevant to the `ImageStream`